src/api/app.py

```diff
@@ -514,6 +515,10 @@ async def fetch_user_from_rw_api(
 
     token = authorization.credentials
 
+    # Handle machine user tokens with zeno-key prefix
+    if token and token.startswith(f"{MACHINE_USER_PREFIX}_"):
```

Comment:
Minor, can we use `:` as the separator instead of `_` - this would be consistent with how we're handling tokens from anonymous users, where we prefix with ANONYMOUS_USER_PREFIX (see code below in same file).